File: src/main/java/hu/frontrider/arcana/blocks/BlockBase.java
package hu.frontrider.arcana.blocks;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;

import static hu.frontrider.arcana.ThaumicArcana.INSTANCE;
import static hu.frontrider.arcana.ThaumicArcana.MODID;

public class BlockBase extends Block {

    public BlockBase(Material materialIn,String name) {
        super(materialIn);
        setRegistryName(MODID,name);
        setUnlocalizedName(MODID +"."+name);
        setCreativeTab(INSTANCE.getTABARCANA());
    }
}

File: src/main/java/hu/frontrider/arcana/blocks/BlockBase.kt
package hu.frontrider.arcana.blocks

import hu.frontrider.arcana.ThaumicArcana.MODID
import hu.frontrider.arcana.ThaumicArcana.TABARCANA
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.ItemStack
import net.minecraft.util.NonNullList

open class BlockBase(materialIn: Material, name: String) : Block(materialIn) {

    init {
        setRegistryName(MODID, name)
        unlocalizedName = "$MODID.$name"
        this.setCreativeTab(TABARCANA)
    }

}
